python/cudf/cudf/_lib/pylibcudf/interop.pyx

```diff
@@ -77,6 +84,21 @@ def from_arrow(pyarrow_object, *, DataType data_type=None):
     raise TypeError("from_arrow only accepts Table and Scalar objects")
 
 
+@from_arrow.register(pa.DataType)
+def _from_arrow_datatype(pyarrow_object):
+    if isinstance(pyarrow_object, pa.Decimal128Type):
+        return DataType(type_id.DECIMAL128, scale=-pyarrow_object.scale)
+    elif isinstance(pyarrow_object, pa.StructType):
+        return DataType(type_id.STRUCT)
+    elif isinstance(pyarrow_object, pa.ListType):
+        return DataType(type_id.LIST)
```

Comment:
My understanding is that the "nestedness" is a property of the column and not as much a property of the type, at least from libcudf's perspective. 

For the `from_arrow` case, there's nothing to return except for `DataType(type_id.LIST)`. For `to_arrow`, we have the column to inspect for the non dtype cases, but not for the dtype only case. Since we can't figure out exactly what kind of `pa.List` or `pa.Struct` to return, I chose to error there.

Reply:
There simply isn't a 1-1 mapping between pyarrow and pylibcudf data types for nested types because in pylibcudf (read: libcudf) the nested types are encoded in the column as Brandon pointed out. So I don't think there's much else we can do in the `from_arrow` case.

For the `to_arrow` case, maybe we should accept a kwarg `column` that can be inspected in the case of list/struct types. WDYT?

Reply:
Walrus (walri?) abound in my attempt here https://github.com/rapidsai/cudf/pull/15257/commits/bffe50014987054cfb84ba21a8c3e64c65e9f007